EIPS/eip-7484.md

```diff
@@ -0,0 +1,272 @@
+
+## Specification
+
+The key words "MUST", "MUST NOT", "REQUIRED", "SHALL", "SHALL NOT", "SHOULD", "SHOULD NOT", "RECOMMENDED", "NOT RECOMMENDED", "MAY", and "OPTIONAL" in this document are to be interpreted as described in RFC 2119 and RFC 8174.
+
+### Definitions
+
+- **Smart account** - An ERC-4337 compliant smart contract account that has a modular architecture.
+- **Module** - Self-contained smart account functionality.
+- **Attestation** - Onchain assertions made about the security of a module.
+- **Attester** - The entity that makes an attestation about a module.
+- **(Module) Registry** - A contract that stores an onchain list of attestations about modules.
+- **Adapter** - Smart account functionality that handles the fetching and validation of attestations from the Registry.
+
+### Required Registry functionality
+
+There are 2 separate required Registry methods: `check` and `checkN`
+
+- `check` is used to check the attestation on a module by a single attester.
+- `checkN` is used to check the attestation on a module by multiple attesters.
+
+The core interface for a Registry is as follows:
+
+```solidity
+interface IRegistry {
+  function check(address module, address attester) public view returns (uint256);
+
+  function checkN(address module, address[] memory attesters, uint256 threshold) external view returns (uint256[] memory);
+}
+```
+
+The Registry MUST implement the following functionality:
+
+- Verify that an attester is the creator of an attestation, for example by checking `msg.sender` or by using signatures, before storing it.
+- Allow attesters to revoke attestations that they have made.
+- Store either the attestation data or a reference to the attestation data.
+- Implement `check` and `checkN` as specified below.
+
+The Registry SHOULD implement the following additional functionality:
+
+- Allow attesters to specify an expiry date for their attestations and revert during `check` or `checkN` if an attestation is expired.
+- Implement a view function that allows an adapter or offchain client to read the data for a specific attestation.
+
+#### `check`
+
+Takes two arguments: `module` and `attester`.
+
+- The Registry MUST revert if the `attester` has not made an attestation on the `module`.
+- The Registry MUST revert if the `attester` has revoked their attestation on the `module`.
```

Comment:
There should be one more point here IMO

The Registry MUST revert if the `attestation` provided is expired.

Same holds for checkN method.

Reply:
Currently, we think it makes more sense to have this be an optional (but encouraged) requirement since some registries might choose to not implement expiration